File: packages/ruby/lib/nurse_andrea/platform_detector.rb
# PRIVACY: This file reads environment variable NAMES to detect
# the hosting platform. Only the platform name is transmitted.
# See DATA_PRIVACY_POLICY.rb for the full policy.

module NurseAndrea
  module PlatformDetector
    PLATFORMS = {
      railway:      -> { ENV.key?("RAILWAY_ENVIRONMENT") },
      render:       -> { ENV.key?("RENDER") },
      fly:          -> { ENV.key?("FLY_APP_NAME") },
      heroku:       -> { ENV.key?("DYNO") },
      digitalocean: -> { ENV.key?("DIGITALOCEAN_APP_PLATFORM_COMPONENT_NAME") },
      vercel:       -> { ENV.key?("VERCEL") }
    }.freeze

    def self.detect
      PLATFORMS.each { |name, check| return name.to_s if check.call }
      "unknown"
    end

    def self.context
      platform = detect
      ctx = { platform: platform }

      case platform
      when "railway"
        ctx[:region] = ENV["RAILWAY_REGION"] if ENV.key?("RAILWAY_REGION")
        ctx[:environment] = ENV["RAILWAY_ENVIRONMENT"] if ENV.key?("RAILWAY_ENVIRONMENT")
      when "render"
        ctx[:region] = ENV["RENDER_REGION"] if ENV.key?("RENDER_REGION")
      when "fly"
        ctx[:region] = ENV["FLY_REGION"] if ENV.key?("FLY_REGION")
      when "heroku"
        ctx[:dyno_type] = ENV["DYNO"]&.gsub(/\.\d+$/, "") if ENV.key?("DYNO")
      end

      ctx
    end
  end
end

File: packages/node/src/platform_detector.ts
// PRIVACY POLICY: Only platform name derived from env var presence.
// No env var values are read or transmitted.

const PLATFORM_SIGNALS = [
  { name: "railway",      vars: ["RAILWAY_ENVIRONMENT", "RAILWAY_SERVICE_NAME"] },
  { name: "render",       vars: ["RENDER", "RENDER_SERVICE_ID"] },
  { name: "fly",          vars: ["FLY_APP_NAME", "FLY_REGION"] },
  { name: "heroku",       vars: ["DYNO", "HEROKU_APP_NAME"] },
  { name: "digitalocean", vars: ["DIGITALOCEAN_APP_NAME"] },
  { name: "vercel",       vars: ["VERCEL", "VERCEL_ENV"] },
] as const

export function detectPlatform(): string {
  for (const { name, vars } of PLATFORM_SIGNALS) {
    if (vars.some(v => v in process.env)) return name
  }
  return "unknown"
}
